### PR TITLE
Adding arm64 and buildx support to the dockerfile and github actions

### LINKED ISSUE #69: Feature: Add support for arm64 `not-a-build-harness`
The need for this feature came around due to iptables not working on emulated amd64 containers, this means that sshuttle won't function. sshuttle uses iptables + nft to configure the tunnel on linux.

I was running into this issue:
https://github.com/multiarch/qemu-user-static/issues/191

After tweaking the dockerfile, I was able to locally build and run`not-a-build-harness` as an arm64 container and was able to successfully use sshuttle and iptables commands.

```bash
docker buildx build --platform linux/arm64/v8,linux/amd64 -t not-a-build-harness:multiarch . --progress plain
```

### PR DESCRIPTION
**Pull request by zack-is-cool**
Resolves #69

### CODE REVIEW

**Comment by zack-is-cool** [comment]
Python took like 30 minutes to install on the arm container 🤔

**Comment by RothAndrew** [comment]
@zack-is-cool As I was reviewing I ran across a much simpler example of doing this. Do you know whether this is something that would work for us? Or is the more complex version as it is here in the PR the only way?

https://github.com/docker/docs/blob/3275456427dbe05c249f3215cefa83ad27219918/build/ci/github-actions/examples.md

File: .github/workflows/release.yml [hunk 1]
```diff
@@ -12,30 +12,83 @@ env:
...
+          username: ${{ github.actor }}
+          password: ${{ secrets.GITHUB_TOKEN }}
+      - name: Create manifest list and push
+        working-directory: /tmp/digests
+        run: |
+          docker buildx imagetools create $(jq -cr '.tags | map("-t " + .) | join(" ")' <<< "$DOCKER_METADATA_OUTPUT_JSON") \
+            $(printf '${{ env.REGISTRY }}/${{ env.IMAGE_NAME }}@sha256:%s ' *)
+      - name: Inspect image
+        run: |
+          docker buildx imagetools inspect ${{ env.REGISTRY }}/${{ env.IMAGE_NAME }}:${{ steps.meta.outputs.version }}
```
Comment (RothAndrew) [inline .github/workflows/release.yml:94]: I'm not sure I'm understanding how it is tagging the different architectures. Can you explain? Does it do something like `X.Y.Z-amd64`?

File: .github/workflows/verify-build.yml [hunk 2]
```diff
@@ -8,25 +8,44 @@ env:
...
-          tags: ${{ steps.meta.outputs.tags }}
+          platforms: ${{ matrix.platform }}
           labels: ${{ steps.meta.outputs.labels }}
-          no-cache: true
+          outputs: type=image,name=${{ env.REGISTRY }}/${{ env.IMAGE_NAME }},push-by-digest=true,name-canonical=true,push=false
+      - name: Export digest
+        run: |
+          mkdir -p /tmp/digests
+          digest="${{ steps.build.outputs.digest }}"
+          touch "/tmp/digests/${digest#sha256:}"
```
Comment (RothAndrew) [inline .github/workflows/verify-build.yml:51]: Are we able to centralize a bunch of this to make it more DRY? It looks like there is a lot of duplication between this file and release.yml

File: .github/workflows/verify-build.yml [hunk 3]
```diff
@@ -8,25 +8,44 @@ env:
...
+        id: build
         uses: docker/build-push-action@v4
         with:
           context: .
-          push: false
-          tags: ${{ steps.meta.outputs.tags }}
+          platforms: ${{ matrix.platform }}
           labels: ${{ steps.meta.outputs.labels }}
-          no-cache: true
+          outputs: type=image,name=${{ env.REGISTRY }}/${{ env.IMAGE_NAME }},push-by-digest=true,name-canonical=true,push=false
```
Comment (RothAndrew) [inline .github/workflows/verify-build.yml:46]: add no-cache, same comment as above.

File: .github/workflows/verify-build.yml [hunk 4]
```diff
@@ -8,25 +8,44 @@ env:
...
+      - name: Export digest
```
Comment (RothAndrew) [inline .github/workflows/verify-build.yml:47]: Do we need to export the digests if we aren't uploading them?

**Review by RothAndrew — CHANGES_REQUESTED** [review]
My only overall concern is that doing this removes the ability to easily switch python versions. However, I don't believe that's something that anyone using this image needs right now, so I'm good with it if it saves tons of time in the build process. If feedback comes out that different python versions need to be used we may need to reevaluate.

**Review by RothAndrew — APPROVED** [review]

**Pull request merged**